File: Master40.SimulationImmutables/Interface/IJob.fs
module IJobs

open FStartConditions
open FProposals
open Akka.Actor
open System
open Master40.DB.DataModel

type public IJob = 
    abstract member Key : Guid with get
    abstract member Name : string with get
    abstract member ForwardStart : int64 with get
    abstract member ForwardEnd : int64 with get
    abstract member BackwardStart : int64 with get
    abstract member BackwardEnd : int64 with get
    abstract member Start : int64 with get
    abstract member End : int64 with get
    abstract member StartConditions : FStartCondition with get
    abstract member Priority : int64 -> double 
    abstract member Proposals : System.Collections.Generic.List<FProposal> 
    abstract member ResourceAgent : IActorRef
    abstract member HubAgent : IActorRef
    abstract member DueTime : int64 with get
    abstract member Duration : int64 with get
    abstract member Tool : M_ResourceTool with get
    abstract member UpdateEstimations : int64 -> IActorRef -> IJob

File: Mate.Production.Core/Environment/Records/Interfaces/IJob.cs
using System;
using Akka.Actor;
using Mate.DataCore.DataModel;

public interface IJob
{
    Guid Key { get; }
    string Name { get; }
    DateTime ForwardStart { get; }
    DateTime ForwardEnd { get; }
    DateTime BackwardStart { get; }
    DateTime BackwardEnd { get; }
    DateTime Start { get; }
    DateTime End { get; }
    StartConditionRecord StartConditions { get; }
    Func<DateTime, double> Priority { get; }
    IActorRef HubAgent { get; }
    DateTime DueTime { get; }
    TimeSpan Duration { get; }
    int SetupKey { get; }
    M_ResourceCapability RequiredCapability { get; }
    Func<DateTime, IJob> UpdateEstimations { get; }
    string Bucket { get; }
    Func<string, IJob> UpdateBucket { get; }
    Action ResetSetup { get; }
}
